pkg/v1/providers/yttcb/clusterbootstrap.yaml

```diff
@@ -0,0 +1,116 @@
+#@           data.values.ANTREA_TRAFFIC_ENCAP_MODE != "encap" or
+#@           data.values.ANTREA_PROXY != True or
+#@           data.values.ANTREA_ENDPOINTSLICE != True or
+#@           data.values.ANTREA_POLICY != True or
+#@           data.values.ANTREA_NODEPORTLOCAL != False or
+#@           data.values.ANTREA_TRACEFLOW != True or
+#@           data.values.ANTREA_EGRESS != False or
+#@           data.values.ANTREA_FLOWEXPORTER != False or
+#@           data.values.ANTREA_DISABLE_UDP_TUNNEL_OFFLOAD != False)
+#@ end
+
+#@ def should_create_clusterbootstrap():
+#@   return data.values.CNI != "antrea" or is_antrea_configs_given()
+#@ end
+
+#@ if should_create_clusterbootstrap():
+---
+apiVersion: run.tanzu.vmware.com/v1alpha3
+kind: ClusterBootstrap
+metadata:
+  name: #@ data.values.CLUSTER_NAME
+  namespace: #@ data.values.NAMESPACE
+  annotations:
+    tkg.tanzu.vmware.com/add-missing-fields-from-tkr: #@ "{}/{}".format(data.values.CLUSTER_NAME, data.values.NAMESPACE)
+spec:
+  cni:
+    #@ if data.values.CNI == "antrea":
+    refName: antrea*
+    valuesFrom:
+      providerRef:
+        apiGroup: cni.tanzu.vmware.com
+        kind: AntreaConfig
+        name: #@ data.values.CLUSTER_NAME
+    #@ elif data.values.CNI == "calico":
+    refName: calico*
+    valuesFrom:
+      providerRef:
+        apiGroup: cni.tanzu.vmware.com
+        kind: CalicoConfig
+        name: #@ data.values.CLUSTER_NAME
+    #@ end
+  kapp:
+    refName: kapp-controller*
+    valuesFrom:
+      providerRef:
+        apiGroup: run.tanzu.vmware.com
+        kind: KappControllerConfig
+        name: #@ data.values.CLUSTER_NAME
+  csi:
+    refName: vsphere*
+    valuesFrom:
+      providerRef:
+        apiGroup: csi.tanzu.vmware.com
+        kind: VSphereCSIConfig
+        name: #@ data.values.CLUSTER_NAME
+#@ end
+
+---
+#@ if is_antrea_configs_given():
+apiVersion: cni.tanzu.vmware.com/v1alpha1
+kind: AntreaConfig
+metadata:
+  name: #@ data.values.CLUSTER_NAME
+  namespace: #@ data.values.NAMESPACE
+spec:
+  antrea:
+    config:
+      #@ if data.values.NSXT_POD_ROUTING_ENABLED:
+      trafficEncapMode: "noEncap"
+      noSNAT: true
+      #@ else:
+      trafficEncapMode: #@ data.values.ANTREA_TRAFFIC_ENCAP_MODE
+      noSNAT: #@ data.values.ANTREA_NO_SNAT
+      #@ end
+      disableUdpTunnelOffload: #@ data.values.ANTREA_DISABLE_UDP_TUNNEL_OFFLOAD
+      featureGates:
+        #@ if data.values.NSXT_POD_ROUTING_ENABLED:
+        AntreaProxy: true
+        #@ else:
+        AntreaProxy: #@ data.values.ANTREA_PROXY
+        #@ end
+        EndpointSlice: #@ data.values.ANTREA_ENDPOINTSLICE
+        AntreaPolicy: #@ data.values.ANTREA_POLICY
+        NodePortLocal: #@ data.values.ANTREA_NODEPORTLOCAL
+        AntreaTraceflow: #@ data.values.ANTREA_TRACEFLOW
+        Egress: #@ data.values.ANTREA_EGRESS
+        FlowExporter: #@ data.values.ANTREA_FLOWEXPORTER
+#@ end
+
+---
+#@ if data.values.CNI == "calico":
+apiVersion: cni.tanzu.vmware.com/v1alpha1
```

Comment:
Leave everything blank should work, but ideally there should be 2 params.
1. `spec.calico.config.vethMTU.skipCNIBinaries: true`
As the pr https://github.com/vmware-tanzu/tanzu-framework/pull/2416 introduced, this param is the fix to stop calico from overwriting the existing cni plugin binaries in the nodes. Because for the OVAs of the providers like vsphere, aws, and azure, the cni plugin binaries are already installed. So this param should be true to skip installing cni plugin binaries on all providers except docker, since docker nodes do not have these binaries.
2. `spec.calico.config.vethMTU: "0"`
Setting this param to "0" or leave it as blank will make the mtu auto detected. If user wants to set a fix mtu value, it can be configured here. But the mtu param is not exposed to the tanzu cli configuration file yet, so I guess we cannot set it based on `data.values`. We have an issue https://github.com/vmware-tanzu/tanzu-framework/issues/1989 to track this mtu expose.

Reply:
Okay, these are either new or not previously exposed so we should be good. This file is for carrying over legacy configs.

Reply:
I saw @vijaykatam mentioned above that [the antrea configurations in 1.6 should be ported here](https://github.com/vmware-tanzu/tanzu-framework/pull/2414/files#r881928085). The `spec.calico.config.skipCNIBinaries` is also a configuration in 1.6 and already set according to the providers. 
https://github.com/vmware-tanzu/tanzu-framework/blob/59a1bc05c28c912054b03043a7009c4aa45243a9/pkg/v1/providers/ytt/02_addons/cni/calico/calico_addon_data.lib.yaml#L11-L12

And the pr https://github.com/vmware-tanzu/tanzu-framework/pull/2416 which contains this `skipCNIBinaries` in CalicoConfig just got merged into main but not in this `package-based-lcm`branch. How should we handle this?

Reply:
Would be a question for @vuil or @saji-pivotal who are doing the rebase.